SYCL/OptionalKernelFeatures/is_compatible.cpp

```diff
@@ -0,0 +1,34 @@
+// requires: cpu
+// RUN: %clangxx -fsycl -O0 %s -o %t.out
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+
+#include <sycl/sycl.hpp>
+
+class Kernel1;
+class Kernel2;
+
+[[sycl::device_has(sycl::aspect::cpu)]] void foo() {};
+[[sycl::device_has(sycl::aspect::gpu)]] void bar() {};
```

Comment:
Using macros, could we change this test so that based on which device we are running it on we pick different aspects here? That way we won't have to limit it to CPU.

Another option is to add a third function here and have the checks below check what kind of device `dev` is and expect support based on that. I.e. for `Kernel1` we would have something like:

```c++
if (sycl::is_compatible<Kernel1>(dev)) {
  ...
  if (!dev.is_cpu()) {
    // Maybe print an error here then return or increment a failure counter.
  }
  ...
}
```

Reply:
Agree, we shouldn't stop on the cpu. What do you think of this version? 536c93e
P.S. I believe some people would like to shorten this code by creating some `test()` function since all three cases are the same, but I'd like to keep it as is to save code simplicity.

Reply:
Yep, that's exactly what I had in mind!
It's not a big test so I don't mind keeping it a little verbose for the sake of readability.